hmda-analytics/src/main/scala/hmda/analytics/HmdaAnalyticsApp.scala

```diff
@@ -392,86 +316,13 @@ object HmdaAnalyticsApp extends App with TransmittalSheetComponent with LarCompo
-                larRepository2020.insert(
-                  LarConverter(lar = lar, 2020)
-                )
-              case Period(2021, None) =>
-                larRepository2021.insert(
-                  LarConverter(lar = lar, 2021)
-                )
-              case Period(2022, None) =>
-                larRepository2022.insert(
-                  LarConverter(lar = lar, 2022)
-                )
-              case Period(2023, None) =>
-                larRepository2023.insert(
-                  LarConverter(lar = lar, 2023)
-                )
-              case Period(2021, Some("Q1")) =>
-                larRepository2021Q1.insert(
-                  LarConverter(lar = lar, 2021, isQuarterly = true)
-                )
-              case Period(2021, Some("Q2")) =>
-                larRepository2021Q2.insert(
-                  LarConverter(lar = lar, 2021, isQuarterly = true)
-                )
-              case Period(2021, Some("Q3")) =>
-                larRepository2021Q3.insert(
-                  LarConverter(lar = lar, 2021, isQuarterly = true)
-                )
-              case Period(2022, Some("Q1")) =>
-                larRepository2022Q1.insert(
-                  LarConverter(lar = lar, 2022, isQuarterly = true)
-                )
-              case Period(2022, Some("Q2")) =>
-                larRepository2022Q2.insert(
-                  LarConverter(lar = lar, 2022, isQuarterly = true)
-                )
-              case Period(2022, Some("Q3")) =>
-                larRepository2022Q3.insert(
-                  LarConverter(lar = lar, 2022, isQuarterly = true)
-                )
-              case Period(2023, Some("Q1")) =>
-                larRepository2023Q1.insert(
-                  LarConverter(lar = lar, 2023, isQuarterly = true)
-                )
-              case Period(2023, Some("Q2")) =>
-                larRepository2023Q2.insert(
-                  LarConverter(lar = lar, 2023, isQuarterly = true)
-                )
-              case Period(2023, Some("Q3")) =>
-                larRepository2023Q3.insert(
-                  LarConverter(lar = lar, 2023, isQuarterly = true)
-                )
-              case Period(2024, Some("Q1")) =>
-                larRepository2024Q1.insert(
-                  LarConverter(lar = lar, 2024, isQuarterly = true)
-                )
-              case Period(2024, Some("Q2")) =>
-                larRepository2024Q2.insert(
-                  LarConverter(lar = lar, 2024, isQuarterly = true)
-                )
-              case Period(2024, Some("Q3")) =>
-                larRepository2024Q3.insert(
-                  LarConverter(lar = lar, 2024, isQuarterly = true)
+              case Period(2018, None) => YearlyLarRepositoryWrapper("2018").getLarRepository.insert(LarConverter2018(lar))
+              case Period(submissionId.period.year, None) => YearlyLarRepositoryWrapper(submissionId.period.year.toString).getLarRepository.insert(
```

Comment:
Yeah this works, on another unit test ticket I'm going to refactor this a bit to demonstrate that. I think we should stick to making this depend on the submissionId for now since this line is a part of reading from that.